implementations/cats/src/main/scala/sttp/client3/impl/cats/CatsMonadAsyncError.scala

```diff
@@ -17,13 +17,11 @@ class CatsMonadAsyncError[F[_]](implicit F: Concurrent[F]) extends MonadAsyncErr
     F.recoverWith(rt)(h)
 
   override def async[T](register: ((Either[Throwable, T]) => Unit) => Canceler): F[T] =
-    F.cancelable(register.andThen(c => F.delay(c.cancel)))
+    F.async_(register.andThen(c => F.delay(c.cancel)))
 
   override def eval[T](t: => T): F[T] = F.delay(t)
 
-  override def suspend[T](t: => F[T]): F[T] = F.suspend(t)
-
   override def flatten[T](ffa: F[F[T]]): F[T] = F.flatten(ffa)
 
-  override def ensure[T](f: F[T], e: => F[Unit]): F[T] = F.guarantee(f)(e)
+  override def ensure[T](f: F[T], e: => F[Unit]): F[T] = F.guaranteeCase(f)(_ => e)
```

Comment:
can use syntax `f.guaranteeCase(_ => e)`, and if you don't suspend side-effects in that `=>`, `f.guarantee(e)`

Reply:
I could not find this syntax. I've tried for all syntax from cats but no luck.